### PR TITLE
Filter and edit functionality done

### PR DESCRIPTION
**Pull request by Developers-codz**
## Files to check

`src/Redux/Reducers/notesSlice.js`
`src/components/eachNote/EachNote.jsx`
`src/components/modal/EditModal.jsx`
`src/components/navbar/Navbar.jsx`
`src/functions/getPrioritySorted.js`
`src/functions/getSortedData.js`
`src/functions/getTagsSortedData.js`

## [Live ](https://notesify-git-filter-and-edit-developers-codz.vercel.app/)

Thank you for reviewing my PR :)

### CODE REVIEW

**Review by vrx29 — APPROVED** [review]
Code looks good. Just one UI suggestion while adding not make ADD button to be highlighted so that it can easily be seen by the user. IT will be better for user experience

**Comment by vishalg8454** [comment]
Noted a crash bug, steps to reproduce
- create new note
- click on edit and write some colored text
- click on update
- the app crashes!

File: src/Redux/Reducers/authSlice.js [hunk 1]
```diff
@@ -30,6 +30,22 @@ export const signup = createAsyncThunk("/auth/signup", async (userDetail) => {
...
+    const encodedToken = localStorage.getItem("token");
+    if (encodedToken) {
+      try {
+        const response = await axios.post("/api/auth/verify", {
+          encodedToken: encodedToken,
+        });
+        console.log(response)
+        return response.data;
+      } catch (error) {
+        console.log(error.response);
```
Comment (vrinda-mahajan) [inline src/Redux/Reducers/authSlice.js]: You can remove console.logs

File: src/components/modal/EditModal.jsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    );
+    dispatch(handleToggleEditModal());
+  };
+
+  return (
+    <>
+      <EditorWrapper style={{ backgroundColor: bgColor }}>
+        <CloseButton onClick={() => dispatch(handleToggleEditModal())}>
+          {" "}
+          X
```
Comment (vrinda-mahajan) [inline src/components/modal/EditModal.jsx:78]: Instead of the letter X try to use an icon.

File: src/components/modal/EditModal.jsx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        </CloseButton>
+
+        <TitleBox
+          type="text"
+          placeholder="Add title ....."
+          style={{ backgroundColor: bgColor }}
```
Comment (vrinda-mahajan) [inline src/components/modal/EditModal.jsx:84]: Try to avoid inline styling.

File: src/pages/home/Home.jsx [hunk 4]
```diff
@@ -6,18 +6,25 @@ import { handleToggleModal,getUserNotes } from "../../Redux/Reducers/notesSlice"
 import { useDispatch } from "react-redux";
 import { EachNote } from "../../components/eachNote/EachNote";
 import { Logo } from "../../assets/icons";
+import {getPrioritySorted} from "../../functions/getPrioritySorted"
+import { getTagsSortedData } from "../../functions/getTagsSortedData";
+import { getSortedData } from "../../functions/getSortedData";
```
Comment (vrinda-mahajan) [inline src/pages/home/Home.jsx]: Suggestion:- You can create an `index.js` file inside the functions folder and import all the functions from that file.

File: src/Redux/Reducers/notesSlice.js [hunk 5]
```diff
@@ -1,18 +1,23 @@
 import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
-import { SuccessToast,AlertToast } from "../../components/toasts";
+import { SuccessToast, AlertToast } from "../../components/toasts";
```
Comment (vrinda-mahajan) [inline src/Redux/Reducers/notesSlice.js:2]: Suggestion:- You can use absolute path instead of relative as it increases the code readability.

**Review by vrinda-mahajan — APPROVED** [review]
Added a few suggestions, other than this everything looks good.

**Comment by Developers-codz** [comment]
> Noted a crash bug, steps to reproduce
> 
> * create new note
> * click on edit and write some colored text
> * click on update
> * the app crashes!

No more crash, fixed that thanks for finding  bugs

File: src/components/modal/EditModal.jsx (hunk 3 above)
Comment (Developers-codz) [reply src/components/modal/EditModal.jsx:84]: Here I am setting bbackground according to the bgColor state that's why

File: src/Redux/Reducers/authSlice.js (hunk 1 above)
Comment (Developers-codz) [reply src/Redux/Reducers/authSlice.js]: sure

File: src/pages/home/Home.jsx (hunk 4 above)
Comment (Developers-codz) [reply src/pages/home/Home.jsx]: Done

**Pull request merged**